In [1]:
# Минимальная совместимость с 2.7
from __future__ import division, print_function
# отключение предупрждений Anaconda
import warnings
warnings.filterwarnings('ignore')

In [2]:
from sklearn.datasets import load_breast_cancer, load_digits
from sklearn.model_selection import cross_val_score
from sklearn.naive_bayes import BernoulliNB, MultinomialNB, GaussianNB

In [12]:
def proceed_task2(dataset_loader):
    '''Loading dataset and crossvalidating naive_bayes classifiers'''
    data, target = dataset_loader(return_X_y=True)
    
    for classificator, name in zip([BernoulliNB, MultinomialNB, GaussianNB],
                                   ['BernoulliNB', 'MultinomialNB', 'GaussianNB']):
        print ('Best result for %s:\t%.3f' %(name, round(cross_val_score(classificator(),
                                                                        X=data,
                                                                        y=target,
                                                                        n_jobs=-1).mean(),
                                                        3)))

In [27]:
#digits
print ('First object from digits dataset:', load_digits().data[0],sep='\n', end='\n\n')

proceed_task2(load_digits)

First object from digits dataset:
[  0.   0.   5.  13.   9.   1.   0.   0.   0.   0.  13.  15.  10.  15.   5.
   0.   0.   3.  15.   2.   0.  11.   8.   0.   0.   4.  12.   0.   0.   8.
   8.   0.   0.   5.   8.   0.   0.   9.   8.   0.   0.   4.  11.   0.   1.
  12.   7.   0.   0.   2.  14.   5.  10.  12.   0.   0.   0.   0.   6.  13.
  10.   0.   0.   0.]

Best result for BernoulliNB:	0.826
Best result for MultinomialNB:	0.871
Best result for GaussianNB:	0.819


In [30]:
#breast cancer
print ('First object from breast_canser dataset:', load_breast_cancer().data[0],sep='\n', end='\n\n')

proceed_task2(load_breast_cancer)

First object from breast_canser dataset:
[  1.79900000e+01   1.03800000e+01   1.22800000e+02   1.00100000e+03
   1.18400000e-01   2.77600000e-01   3.00100000e-01   1.47100000e-01
   2.41900000e-01   7.87100000e-02   1.09500000e+00   9.05300000e-01
   8.58900000e+00   1.53400000e+02   6.39900000e-03   4.90400000e-02
   5.37300000e-02   1.58700000e-02   3.00300000e-02   6.19300000e-03
   2.53800000e+01   1.73300000e+01   1.84600000e+02   2.01900000e+03
   1.62200000e-01   6.65600000e-01   7.11900000e-01   2.65400000e-01
   4.60100000e-01   1.18900000e-01]

Best result for BernoulliNB:	0.627
Best result for MultinomialNB:	0.895
Best result for GaussianNB:	0.937


## Выводы:

1) максимальное качество на **breast_canser** 0.937

2) максимальное качество на **digits** 0.871

3) c, d